src/lib/configs.js

```diff
@@ -12,25 +13,44 @@ export async function setConfigs(config) {
   GLEE_LIFECYCLE_DIR = path.resolve(GLEE_DIR, config.functionsDir || 'lifecycle')
   GLEE_FUNCTIONS_DIR = path.resolve(GLEE_DIR, config.functionsDir || 'functions')
   GLEE_CONFIG_FILE_PATH = path.resolve(GLEE_DIR, 'glee.config.js')
-  ASYNCAPI_FILE_PATH = path.resolve(GLEE_DIR, 'asyncapi.yaml')
+  ASYNCAPI_FILE_PATH = resolveAsyncApiPath(GLEE_DIR)
   await loadConfigsFromFile()
-  
+
   return getConfigs()
 }
 
+/**
+ * Tries to resolve the path to a valid asyncapi file in the baseDir
+ */
+export async function resolveAsyncApiPath(baseDir) {
+  const asyncApiFiles = ['asyncapi.yaml', 'asyncapi.json', 'asyncapi.yml']
+
+  const promiseArray = asyncApiFiles.map(async (filename) => {
+    return access(path.resolve(baseDir, filename), constants.R_OK)
+      .then(() => { return path.resolve(baseDir, filename) })
+  })
+
+  const fileResults = await Promise.allSettled(promiseArray)
+  const accessiblePaths = fileResults.filter((result) => result.status === 'fulfilled')
+  if (accessiblePaths.length === 0) { throw new Error('AsyncApi file was not found or is not readable') }
+  if (accessiblePaths.length > 1) { throw new Error('Multiple AsyncApi files were found - please remove all but one') }
```

Comment:
```suggestion
  if (accessiblePaths.length > 1) { throw new Error('Multiple AsyncAPI files were found - please remove all but one') }
```

Regarding the message, I think we shouldn't advocate for removing files but to encourage devs to configure Glee to pick one of them. Maybe using the glee.config.js file. Happy if you want to give it a try in a different PR or here. As you prefer.